core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -91,6 +91,7 @@ private[spark] object JsonProtocol {
         executorRemovedToJson(executorRemoved)
       // These aren't used, but keeps compiler happy
       case SparkListenerExecutorMetricsUpdate(_, _) => JNothing
+      case SparkListenerBlockUpdate(_, _, _) => JNothing
```

Comment:
This event should be converted to json so it can be seen in the UI for completed apps.  You need to implement serialization here, and also deserialization in `sparkEventFromJson`, as that is used by `ReplayListenerBus` to show the UI for completed apps.